data-prepper-expression/src/main/java/org/opensearch/dataprepper/expression/DateTimeFormatExpressionFunction.java

```diff
@@ -0,0 +1,87 @@
+package org.opensearch.dataprepper.expression;
+
+import java.time.Instant;
+import java.time.ZoneId;
+import java.time.format.DateTimeFormatter;
+import java.time.temporal.TemporalAccessor;
+import java.util.List;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+import javax.inject.Named;
+import org.opensearch.dataprepper.model.event.Event;
+
+@Named
+public class DateTimeFormatExpressionFunction implements ExpressionFunction{
+    @Override
+    public String getFunctionName() {
+        return "dateTimeFormat";
```

Comment:
Let's call it `formatDateTime` to match our other functions which start with a verb.